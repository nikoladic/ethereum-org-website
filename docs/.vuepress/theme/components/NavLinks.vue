<template>
  <nav class="nav-links" v-if="userLinks.length">
    <ul class="nav-ul no-list">
      <li
        class="nav-item no-list mr-1"
        v-for="item in userLinks"
        :key="item.link"
      >
        <DropdownLink v-if="item.type === 'links'" :item="item" />
        <NavLink v-else :item="item" />
      </li>
      <li class="nav-item" v-if="isSidebar">
        <router-link class="nav-link" to="/languages/">Languages</router-link>
      </li>
    </ul>
  </nav>
</template>

<script>
import { isActive, resolveNavLinkItem } from '../utils/util'
import { translate } from '../utils/translations'
import NavLink from './NavLink.vue'
import DropdownLink from './DropdownLink.vue'

export default {
  components: { NavLink, DropdownLink },
  props: ['isSidebar'],
  computed: {
    nav() {
      const languagePath = translate('path', this.$lang)
      return this.$site.locales[languagePath].nav || []
    },
    userLinks() {
      return (this.nav || []).map(link => {
        return Object.assign(resolveNavLinkItem(link), {
          items: (link.items || []).map(resolveNavLinkItem)
        })
      })
    }
  },
  methods: {
    isActive
  }
}
</script>

<style lang="stylus">
@import '../styles/config.styl';

.nav-ul
  margin 0
  display flex
  list-style-type none
  list-style-image none
  @media (max-width $breakS)
    display block

  & > li
    color $subduedColor
    margin-right 1em

.nav-links
  display inline-block
  vertical-align top

  .nav-item
    cursor pointer
    position relative
    font-weight 500

@media (max-width $MQMobile)
  .nav-links
    .nav-item
      margin-left 0
</style>
