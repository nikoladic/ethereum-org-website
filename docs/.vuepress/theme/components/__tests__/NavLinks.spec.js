import { shallowMount } from '@vue/test-utils'
import NavLinks from "@/theme/components/NavLinks"

jest.mock("../../../theme/utils/translations");

describe('NavLinks', () => {

  const mocks = {
    $site: {
      locales: {
        "/": {
          nav: [{
            items: ["en-US", "en-GB"],
            link: "/"
          }],
        },
        "/de/": {
          nav: "de",
          link: "/de/"
        }
      },
    },
    $lang: "en-US"
  }

  test('render as sidebar, with no navigation', () => {
    const wrapper = shallowMount(NavLinks, {
      propsData: {
        isSidebar: true
      },
      mocks: {
        $site: {
          locales: {
            "/": {}
          },
        },
        $lang: "en-US"
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('render as sidebar, with navigation and sub-items', () => {
    const wrapper = shallowMount(NavLinks, {
      propsData: {
        isSidebar: true
      },
      mocks
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('render not as sidebar', () => {
    const wrapper = shallowMount(NavLinks, {
      propsData: {
        isSidebar: false
      },
      mocks
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
