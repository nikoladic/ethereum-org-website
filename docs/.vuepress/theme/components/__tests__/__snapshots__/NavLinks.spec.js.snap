// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinks render as sidebar, with navigation and sub-items 1`] = `
<nav
  class="nav-links"
>
  <ul
    class="nav-ul"
  >
    <li
      class="nav-item"
    >
      <dropdownlink-stub
        item="[object Object]"
      />
    </li>
     
    <languagedropdown-stub
      class="nav-item"
    />
  </ul>
</nav>
`;

exports[`NavLinks render as sidebar, with no navigation 1`] = `<!---->`;

exports[`NavLinks render not as sidebar 1`] = `<!---->`;
